src/Numerics/ODESolvers/DifferentialEquations.jl

```diff
@@ -0,0 +1,115 @@
+import DiffEqBase
+export DiffEqJLSolver, DiffEqJLIMEXSolver
+
+abstract type AbstractDiffEqJLSolver <: AbstractODESolver end
+
+"""
+    DiffEqJLSolver(f, RKA, RKB, RKC, Q; dt, t0 = 0)
+
+This is a time stepping object for explicitly time stepping the differential
+equation given by the right-hand-side function `f` with the state `Q`, i.e.,
+
+```math
+  \\dot{Q} = f(Q, t)
+```
+
+via a DifferentialEquations.jl DEAlgorithm, which includes support
+for OrdinaryDiffEq.jl, Sundials.jl, and more.
+"""
+mutable struct DiffEqJLSolver{I} <: AbstractDiffEqJLSolver
+    integ::I
+
+    function DiffEqJLSolver(rhs!,alg,Q,args...;
+                            t0 = 0, p = nothing,
+                            kwargs...)
+        prob = DiffEqBase.ODEProblem(
+                          (du,u,p,t)->rhs!(du,u,p,t; increment=false),
+                           Q, (float(t0),typemax(typeof(float(t0)))),
+                           p)
+        integ = DiffEqBase.init(prob,alg,args...;kwargs...)
+        new{typeof(integ)}(integ)
+    end
+end
+
+"""
+    DiffEqJLSolver(f, RKA, RKB, RKC, Q; dt, t0 = 0)
+
+This is a time stepping object for explicitly time stepping the differential
+equation given by the right-hand-side function `f` with the state `Q`, i.e.,
+
+```math
+  \\dot{Q} = f_I(Q, t) + f_E(Q, t)
+```
+
+via a DifferentialEquations.jl DEAlgorithm, which includes support
+for OrdinaryDiffEq.jl, Sundials.jl, and more.
+"""
+mutable struct DiffEqJLIMEXSolver{I} <: AbstractDiffEqJLSolver
+    integ::I
+
+    function DiffEqJLIMEXSolver(rhs!,rhs_implicit!,alg,Q,args...;
+                            t0 = 0,
+                            kwargs...)
+        prob = SplitODEProblem(
+                          (du,u,p,t)->rhs_implicit!(du,u,p,t; increment=false),
+                          (du,u,p,t)->rhs!(du,u,p,t; increment=false),
+                           Q, (float(t0),typemax(typeof(float(t0)))),
+                           p)
+        integ = DiffEqBase.init(prob,alg,args...;kwargs...)
+        new{typeof(integ)}(integ)
+    end
+end
+
+gettime(solver::AbstractDiffEqJLSolver) = solver.integ.t
+getdt(solver::AbstractDiffEqJLSolver) = solver.integ.dt
+updatedt!(solver::AbstractDiffEqJLSolver, dt) = DiffEqBase.set_proposed_dt!(solver.integ,dt)
+updatetime!(solver::AbstractDiffEqJLSolver, t) = DiffEqBase.set_t!(solver.integ,t)
+isadjustable(solver::AbstractDiffEqJLSolver) = true # Is this isadaptive? Or something different?

```

Comment:
`isadjustable` here mainly deals with ode solvers with an implicit component (IMEX RK for example). We currently don't support adaptive time-stepping. It has less to do with the overall `ODESolver` and more to do with the linear solver for the implicit part (you can see how it's used in `src/ODESolvers/BackwardEulerSolvers.jl`.

It's basically a flag determining whether or not the solver can/should be "rebuilt" for changing `dt`. An example would be using an IMEX Runge-Kutta method with different stage time-steps. If you're using, for example, and LU factorization to solve the implicit part, then `isadjustable` will help decide whether or not the LU factors are built only once or during each stage update.

I think for now, what you have here is fine. This is something we need to look more carefully at and try to clean up. IIRC, a solver being adjustable is determined by the `LinearBackwardEulerSolver` passed to the IMEX method.

@jkozdon can probably correct me if I am wrong on this.

Reply:
>It's basically a flag determining whether or not the solver can/should be "rebuilt" for changing dt. 

Oh, DiffEq is all safe with changing `dt`, so we can just set it to `false`. LU-factorization reduction via Jacobian reuse is one thing we'll need to look into though.